src/mlpack/methods/ann/layer/nearest_interpolation.hpp

```diff
@@ -21,35 +22,38 @@ namespace mlpack {
  * scaling purposes. The input should be a 2D matrix and it can have
  * a number of channels/units.
  *
- * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ * @tparam InputType Type of the input data (arma::colvec, arma::mat,
  *         arma::sp_mat or arma::cube).
- * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ * @tparam MatType Type of the output data (arma::colvec, arma::mat,
  *         arma::sp_mat or arma::cube).
  */
-template <
-    typename InputDataType = arma::mat,
-    typename OutputDataType = arma::mat
->
-class NearestInterpolation
+template<typename MatType = arma::mat>
+class NearestInterpolationType : public Layer<MatType>
 {
  public:
   //! Create the NearestInterpolation object.
-  NearestInterpolation();
+  NearestInterpolationType();
 
-  /**
-   * The constructor for the NearestInterpolation.
-   *
-   * @param inRowSize Number of input rows.
-   * @param inColSize Number of input columns.
-   * @param outRowSize Number of output rows.
-   * @param outColSize Number of output columns.
-   * @param depth Number of input slices.
-   */
-  NearestInterpolation(const size_t inRowSize,
-                       const size_t inColSize,
-                       const size_t outRowSize,
-                       const size_t outColSize,
-                       const size_t depth);
+  NearestInterpolationType(const size_t inRowSize,
+                           const size_t inColSize,
```

Comment:
One of the things about the transition to the new layer interface is that we no longer need to specify the input size in the constructor---so you can remove these parameters, and then use `inputDimensions` inside of `ComputeOutputDimensions()` (it will be set automatically before `ComputeOutputDimensions()` to hold the output size of the previous layer).